### PR TITLE
OOP school library: unit tests

### PR DESCRIPTION
**Pull request by zhorabay**
- [ ] In this branch, we:
- Added unit tests for all the methods in all the classes:
-> Book
-> Person
-> Student
-> Teacher
-> Classroom
-> Rental
-> Decorator
-> Capitalize
-> Trimmer

### CODE REVIEW

File: spec/student_spec.rb [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  end
+
+  it 'should add itself to the classroom' do
+    expect(classroom.students).to include(student)
+  end
+
+  it 'should not add itself to the classroom if already present' do
+    student.classroom = classroom
+    expect(classroom.students.count(student)).to eq(1)
+  end
```
Comment (danmainah) [inline spec/student_spec.rb:25]: - [ ] @Team Great work adding tests for the student class, however, you forgot to add tests for the `play_hooky` method, kindly consider adding a test for this method.

File: spec/trimmer_spec.rb [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+    it 'should wrap a Nameable object' do
+      expect(decorator_long.nameable).to eq(nameable_long)
+    end
+
+    it 'should return a truncated correct name' do
+      expect(decorator_long.correct_name).to eq('ThisIsALon')
+    end
+  end
+end
```
Comment (danmainah) [inline spec/trimmer_spec.rb:33]: - [ ] @Team Great work adding tests for the decorator classes but you forgot to add tests for the `Nameable` class, kindly consider adding tests for this class.

**Review by danmainah — CHANGES_REQUESTED** [review]
Hi @zhorabay and @mryeminthu , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
✔️ Added tests for all the required classes.
✔️ Ensured all tests added are passing.

# Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Review by zeff96 — APPROVED** [review]
### Status: APPROVED :green_circle: 
Hi @zhorabay and @mryeminthu , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
congrats

## Highlights
:heavy_check_mark: Linters passing
:heavy_check_mark: All test implemented for all methods


## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**